_posts/2020-05-22-smallrye-config-1-8.adoc

```diff
@@ -0,0 +1,155 @@
+loaded the resolved config:
+
+[source,java]
+----
+public class LoggingConfigSourceInterceptor implements ConfigSourceInterceptor {
+    private static final Logger LOG = Logger.getLogger("io.smallrye.config");
+
+    @Override
+    public ConfigValue getValue(final ConfigSourceInterceptorContext context, final String name) {
+        final ConfigValue configValue = context.proceed(name);
+        final String key = configValue.getName();
+        final String value = configValue.getValue();
+        final String configSource = configValue.getConfigSourceName();
+
+        LOG.infov("The key {0} was loaded from {1} with the value {2}", key, configSource, value);
+
+        return configValue;
+    }
+}
+----
+
+SmallRye Config does ship with its own logging interceptor, so you don't need to implement one.
+
+== Built-In Interceptors
+
+SmallRye Config provides the following built-in Interceptors to expand its feature set:
+
+* `RelocateConfigSourceInterceptor`
+* `ProfileConfigSourceInterceptor`
+* `ExpressionConfigSourceInterceptor`
+* `FallbackConfigSourceInterceptor`
+* `LoggingConfigSourceInterceptor`
+* `SecretKeyConfigSourceInterceptor`
+
+Not every interceptor is registered by default. Only the `ProfileConfigSourceInterceptor`,
+`ExpressionConfigSourceInterceptor` and `SecretKeyConfigSourceInterceptor` are added on new instances of `Config`.
+Other interceptors require manual registration via the `ServiceLoader` mechanism.
+
+To learn more about all the build-in interceptors, please refer to the
+https://smallrye.io/docs/smallrye-config/interceptors/interceptors.html#_built_in_interceptors[documentation page].
+
+=== Expression Expansion
+
+The `ExpressionConfigSourceInterceptor` provides expression expansion on configuration values. An expression string is
+a mix of plain strings and expression segments, which are wrapped by the sequence `${ ... }`.
+
+For instance, the following configuration properties file:
+
+[source,properties]
+----
+my.prop=1234
+expression=${my.prop}
+----
+
+Then the `expression` configuration will be resolved and expanded to the value `1234`.
+
+Additionally, the expression expansion engine supports the following segments:
+
+* `${expression:value}` - Provides a default value after the `:` if the expansion doesn't find a value.
+* `${my.prop${compose}}` - Composed expressions. Inner expressions are resolved first.
+* `${my.prop}${my.prop}` - Multiple expressions.
+
+=== Configuration Profiles
+
+The `ProfileConfigSourceInterceptor` allows multiple configurations with the same name and selects them via
+a profile property.
+
+To be able to set properties with the same name, each property needs to be prefixed with `%` followed by the profile
+name and a dot:
+
+[source,properties]
+----
+my.prop=1234
+%dev.my.prop=5678
+----
+
+Lookup is always performed with the `my.prop` property name. To use the profile `dev`, the configuration
+`smallrye.config.profile=dev` has to be set into any valid `ConfigSource`.
```

Comment:
"has to be set into any valid `ConfigSource`" looks weird, perhaps just "has to be set in any valid `ConfigSource`"?